### DISCUSSION TITLE
Does ebpf-go support BPF_PROG_TYPE_PERF_EVENT?

### DISCUSSION THREAD

**Discussion by caneatcansleep** [cilium/ebpf#946] [Q&A]
Hi, I want to know whether ebpf-go support [BPF_PROG_TYPE_PERF_EVENT](https://github.com/cilium/ebpf/blob/8c2039b96a96db0a60f9a897858c5f0fb70eb18f/types.go#L129), currently the [attachPerfEvent](https://github.com/cilium/ebpf/blob/8c2039b96a96db0a60f9a897858c5f0fb70eb18f/link/perf_event.go#L188)  function only support Kprobe, TracePoint,  and I did't find  any cases or examples about BPF_PROG_TYPE_PERF_EVENT.

**Comment by ti-mo** [comment]
What is this program type used for? A grep through the kernel selftests turns up some things related to periodic perf events, but I don't really find any examples. There was a PR to add periodic events (#924) to the perf package, but we're not going to add those in the near future. We're considering spinning off the perf package into a separate repo to allow the community to broaden its scope a bit.

What is your use case?
